doc/components.md

```diff
@@ -0,0 +1,221 @@
+# Components
+
+This document provides a full introduction and usage example of each component implemented here. Let's start with it!
+
+## config/aero
+This component is responsible for reading and parsing configuration files from our environment based on a `resources/config.edn` file or getting the current [profile](https://github.com/juxt/aero#profile) on environment var `SYSTEM_ENV`. The used library is defined as `A small library for explicit, intentful configuration` and you can configure multiple environment variables for each specific usage.
+
+Look below for an example of `resources/config.edn` file:
+```clojure
+{:webserver/port #long #or [#env PORT 3001]
+
+ :database {:dbtype "postgres"
+            :dbname #or [#env DB-NAME "postgres"]
+            :username #or [#env DB-USER "postgres"]
+            :password #or [#env DB-PASS "postgres"]
+            :host #or [#env DB-HOST "localhost"]
+            :port #or [#env DB-PORT 5432]}}
+```
+We define a `:webserver/port` for example, that can come from a defined `PORT` in our environment, or if it's not defined, it will assume as `3001`. The same logic is valid for each field in the database configuration.
+
+To start our component we can start building our system map:
+```clojure
+(defn- build-system-map []
+  (component/system-map
+   ;; some other components
+   :config (config/new-config)))
+```
+> It's normal to see this component as the first one on the system map because it will start with every configuration that comes from our environment.
+
+The `dev` profile is the default for `SYSTEM_ENV`, so if you want to set up another profile you can define your `SYSTEM_ENV` to a new value and a new environment will be configured.
```

Comment:
Since you mentioned profile here, maybe it's interesting to have an example of how use it in aero

https://github.com/juxt/aero?tab=readme-ov-file#profile